ouroboros-consensus/docs/report/chapters/future/genesis.tex

```diff
@@ -1499,3 +1500,178 @@ \subsection{Possible optimisations}
+chains of \emph{headers}. These headers provide us with sufficient context to do
+chain selection and pick the best chain\footnote{Technically speaking this is a
+preliminary chain selection, because the chain we chose might contain invalid
+blocks. If this is the case, we will notice when we download the blocks and
+attempt to actually adopt the chain in the chain database. We will then
+disconnect from the peer that served us the offending chain, at which point a
+different chain might now be the preferred chain. This introduces a delay in
+selecting the optimal valid chain, but we accept this risk as sufficiently small
+that we do not pre-emptively download blocks from all chains. After all, there
+is little incentive for an attacker with the right to produce blocks to produce
+\emph{invalid} blocks.}. Under normal circumstances this is the \emph{only}
+chain we will download the blocks of and adopt, but occassionally we might want
+to download blocks from chains \emph{other} than the best chain, perhaps because
+downloading that best chain is taking a very long time (it might even be the
+case that the remote node is intentionally trying to stall us, as a form of
+denial of service attack). We will refer to this as \emph{suboptimal chain
+selection}; in this section we will discuss the conditions under which this is
+sound. Note that a simple argument that we might anyway not yet have seen the
+optimal chain does not hold water, due to the $\Delta$ guarantee.
+
+TODO\todo{TODO}: Should we think harder about pre-emptively download blocks?
+After all, a node giving us invalid blocks is \emph{also} introducing delays,
+which is precisely the kind of thing we are worried about here.
+One difference is that with an invalid block, we will disconnect; however,
+nothing is stopping a node from doing this again and again.
+I see two possible solutions: pre-emptively download blocks after all, or
+(more appealing) associate invalid blocks with signer keys?
+
+\subsection{Fundamental principles}
+
+We will take as our starting point the following two fundamental principles:
+
+\begin{description}
+\item[No unnecessary delays]
+The propagation of the best chain must not be unnecessarily delayed. Since in
+our implementation nodes only propagate their \emph{own} chain, this means that
+the \emph{adoption} of the best chain must not be unnecessarily delayed.
+
+\item[The network respects $\Delta$]
+It is not in fact very clear how the value of $\Delta$ as it is used in the
+Ouroboros paper relates to the implementation, which differs from the paper
+rather significantly. However, what we \emph{can} say is that when we get to
+forge a block, \emph{either} we received all blocks we should have received
+``within $\Delta$'', in which case forging that block on top of the best chain
+we have available matches the paper; \emph{or} some blocks got delayed ``more
+than $\Delta$'', and we end up forging on the ``wrong'' chain. It can not,
```

Comment:
I don't think using `wrong` here is the right word. If we didn't receive the optimal one within delta, then the "virtual optimal" chain is the one that we previously considered suboptimal, so it is not "the wrong chain", at least not for us.